mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1234,6 +1239,60 @@ public ParameterMetadataBuilder getParameterMetadataBuilder() {
 		return NoOpParameterMetadataBuilder.INSTANCE;
 	}
 
+	@Override
+	public void createIndex(OgmIndexSpec ogmIndexSpec) {
+		MongoDBIndexSpec indexSpec = (MongoDBIndexSpec) ogmIndexSpec;
+		DBCollection collection = getCollection( indexSpec.getCollection() );
+		Map<String, DBObject> preexistingIndexes = getIndexes( collection );
+		String preexistingTextIndex = getPreexistingTextIndex( preexistingIndexes );
+
+		// if a text index already exists in the collection, MongoDB silently ignores the creation of the new text index
+		// so we might as well log a warning about it
+		if ( indexSpec.isTextIndex() && preexistingTextIndex != null && !preexistingTextIndex.equalsIgnoreCase( indexSpec.getIndexName() ) ) {
+			log.unableToCreateTextIndex( collection.getName(), indexSpec.getIndexName(), preexistingTextIndex );
+			return;
+		}
+
+		try {
+			// if the index is already present and with the same definition, MongoDB simply ignores the call
+			// if the definition is not the same, MongoDB throws an error, except in the case of a text index
+			// where it silently ignores the creation
+			collection.createIndex( indexSpec.getIndexKeysDBObject(), indexSpec.getIndexOptionsDBObject() );
+		}
+		catch (MongoException e) {
+			String indexName = indexSpec.getIndexName();
+			if ( e.getCode() == 85
```

Comment:
It's the MongoDB error code. It's not very precise as it seems it's the same for every error directly linked to the index creation.